### PR TITLE
os/bluestore: drop (semi-broken) 'nvme' automatic class

### PR DESCRIPTION
**Pull request by liewegas**
The SSD vs nvme distinction is weak to useless, and there are
HDDs with nvme interfaces coming in the future.

AFAICS this only works when bluestore is running on the raw nvme
block device--it doesn't work through LVM, which means it isn't
used on most clusters.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by jdurgin** [comment]
Is the NVMEDevice class still useful? it always sets rotational=0

**Comment by tchaikov** [comment]
@jdurgin i think NVMEDevice is still useful. it uses spdk's nvme library to talk to the underlying nvme device.

**Comment by liewegas** [comment]
List users seem to confirm that in practice the 'nvme' class doesn't show up (with ceph-volume).

File: PendingReleaseNotes [hunk 1]
```diff
@@ -311,3 +311,10 @@
   its orignal behaviour of indicating an output file. This reverts it to a more
   consisten behaviour when compared to other tools. Specifying obect size is now
   accomplished by using an upper case O ``-O``.
+
+* In certain rare cases, OSDs would self-classify themselves as type
+  'nvme' instead of 'hdd' or 'ssh'.  This appears to be limited to
```
Comment (votdev) [inline PendingReleaseNotes]: s/ssh/ssd/

**Pull request merged**